.github/workflows/model_ci.yml

```diff
@@ -42,3 +42,119 @@ jobs:
+
+        helm upgrade --install carbon-assistant rasa-x/rasa-x --namespace carbon-assistant \
+            --set global.postgresql.postgresqlPassword="${DB_PASSWORD}" \
+            --set app.tag=$(basename ${{ github.ref }})-${{ github.sha }} \
+            --set rasax.initialUser.password="${RASA_X_ADMIN_PASSWORD}" \
+            --set rasax.token=${RASA_TOKEN} \
+            --set rasax.jwtSecret=${JWT_SECRET} \
+            --set rasa.token=${RASA_X_TOKEN} \
+            --set rasax.passwordSalt=${PASSWORD_SALT} \
+            --set global.redis.password=${REDIS_PASSWORD} \
+            --set rabbitmq.rabbitmq.password=${RABBITMQ_PASSWORD} \
+            --set rasax.disableTelemetry=True \
+            --set rasax.tag="$RASA_X_VERSION" \
+            --set eventService.tag="${RASA_X_VERSION}" \
+            --set nginx.tag="${RASA_X_VERSION}" \
+            --set rasa.tag="${RASA_VERSION}-full" \
+            --values helm-values.yml
+
+        cat <<EOF | kubectl apply --namespace carbon-assistant -f  -
+        apiVersion: networking.gke.io/v1beta1
+        kind: ManagedCertificate
+        metadata:
+          generation: 1
+          name: rasa-bots-certificate
+        spec:
+          domains:
+          - carbon.rasa-bots.com
+        EOF
+
+  upload-model:
+    name: Upload the trained model to Rasa X
+    needs: 
+    - deploy-to-cluster
+    - build-model
+    if: github.ref == 'refs/heads/master'
+    runs-on: ubuntu-latest
+    steps:
+    - name: Download Model
+      uses: actions/download-artifact@master
+      with:
+        name: model
+    - name: Upload Model to Rasa
+      env:
+        RASA_X_PASSWORD: ${{ secrets.RASA_X_ADMIN_PASSWORD }}
+        RASA_X_USERNAME: admin
+      run: |
+        sleep 240
+        RASA_X_DEPLOYMENT_LABEL="app.kubernetes.io/component=rasa-x"
+        # Wait for deployment to be ready
+        kubectl wait --for=condition=available --timeout=420s -l "${RASA_X_DEPLOYMENT_LABEL}" deployment --namespace carbon-assistant
+        # Get rasa-x pod
+        RASA_X_POD=$(kubectl get pods --namespace carbon-assistant -l "${RASA_X_DEPLOYMENT_LABEL}" --no-headers | head -1 | awk '{print $1}')
+        # Get token
+        RASA_X_TOKEN=$(curl -s --header "Content-Type: application/json" \
+          --request POST \
+          --data "{\"username\":\"${RASA_X_USERNAME}\",\"password\":\"${RASA_X_PASSWORD}\"}" \
+          https://carbon.rasa-bots.com/api/auth | jq -r .access_token)
+        # Upload model
+        cd model; export MODELNAME=$(ls); curl -k --fail -H "Authorization: Bearer ${RASA_X_TOKEN}" -F "model=@${MODELNAME}" http://carbon.rasa-bots.com/api/projects/default/models
+        # ensure model is ready and tag as production
+        sleep 5
+        export MODEL=$(ls | xargs -I{} basename {} .tar.gz);curl --fail -XPUT -H "Authorization: Bearer ${RASA_X_TOKEN}" https://carbon.rasa-bots.com/api/projects/default/models/${MODEL}/tags/production
```

Comment:
Looks good 👍 
One very minor thing: maybe it's worth saving a domain name to a constant?